fastai/callbacks/flat_cos_anneal.py

```diff
@@ -0,0 +1,37 @@
+"Supports flat-cosine-annealling style training"
+
+from ..core import *
+from ..callback import *
+from ..basic_train import Learner, LearnerCallback
+
+__all__ = ['FlatCosAnnealScheduler']
+
+# The brain child of Mikhail Grankin aimed for use of the new optimizers
+# For more information: https://forums.fast.ai/t/how-we-beat-the-5-epoch-imagewoof-leaderboard-score-some-new-techniques-to-consider/53453
+
+class FlatCosAnnealScheduler(LearnerCallback):

```

Comment:
I think you should make this a regular function that returns `GeneralScheduler(learn, phases)`. Keep the name capitalized for consistency with the other callbacks.

Second comment: please keep the docstring to one line and add any other info in the doc notebooks when you add the documentation. The link and citation to you are already in the source code in the commented lines 9 and 10, so we don't need it here. You should absolutely put them again in the documentation of this when you make that second PR.

Reply:
Adjusted it so.